### PR TITLE
labels for in progress files

### PR DESCRIPTION
**Pull request by mo271**
Adding labels in the port status as suggested here: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/port.20progress/near/324894844

### CODE REVIEW

**Comment by mo271** [comment]
mathport_with_labels

**Comment by mo271** [comment]
Currently this looks still pretty ugly 🤷

- [x] Make it more pretty (by using the same style for the labels used on github, with rounded corners, white font if label dark etc...)
- [x] Consider authentication if we ever get a 403 when querying github
- [x] Consider making the labels more useful by putting them in their own column

File: make_html.py [hunk 1]
```diff
@@ -17,6 +20,17 @@
...
+@functools.cache
+def github_labels(pr):
+    url = f'https://api.github.com/repos/leanprover-community/mathlib4/pulls/{pr}'
+    response = requests.get(url)
+    json_response = response.json()
+    if 'labels' not in json_response:
+        logging.error(f'curl of github failed for {pr=}, API rate limit exceeded?')
+        return []
+    labels = [{'name': label['name'], 'color': label['color']} for label in json_response['labels']]
+    return labels
```
Comment (eric-wieser) [inline make_html.py]: It probably makes sense to use the python github API here, like we do in https://github.com/leanprover-community/mathlib4/blob/master/scripts/make_port_status.py

File: templates/index.j2 [hunk 2]
```diff
@@ -63,7 +63,11 @@
                     <td>
                         <a href="https://github.com/leanprover-community/mathlib4/pull/{{f_data.status.mathlib4_pr}}">#{{f_data.status.mathlib4_pr}}</a>
                     </td>
-                    <td>{{f_data.status.comments | htmlify_comment if f_data.status.comments}}</td>
+                    <td>{{f_data.status.comments | htmlify_comment if f_data.status.comments}}
+                    {%- for label in f_data.labels %}
+                    <div style="background-color:#{{label.color}};display: inline">{{label.name}}</div>
```
Comment (eric-wieser) [inline templates/index.j2]:
```suggestion
                    <span class="badge rounded-pill" style="background-color:#{{label.color}}">{{label.name}}</span>
```
will give nicer styling.

File: make_html.py (hunk 1 above)
Comment (mo271) [reply make_html.py]: done in very similar ways to what is done in make_port_status.py. I guess the CI just needs to somehow make sure to have a proper token in the `github-token` file

**Comment by mo271** [comment]
Nicer already. 
To get the labels just like they are on github one could copy the css from there, I suppose. Currently the yellow label is a bit hard to read, a dark font would be better here.

roundlabels

File: requirements.txt [hunk 3]
```diff
@@ -2,3 +2,4 @@ jinja2
 mathlibtools
 networkx
 gitpython
+requests
```
Comment (eric-wieser) [inline requirements.txt]: This is no longer used, right? Instead, we need the GitHub API here

File: templates/index.j2 [hunk 4]
```diff
@@ -49,6 +49,7 @@
                     <th>File</th>
                     <th>PR</th>
                     <th>comments</th>
+                    <th>labels</th>
```
Comment (eric-wieser) [inline templates/index.j2:52]: Note that above we sort by column 3, but you inserted a new column

File: make_html.py [hunk 5]
```diff
@@ -17,6 +21,21 @@
 
 from htmlify_comment import htmlify_comment
 
+if not os.path.exists('build/repos/mathlib'):
+    print("Make sure you are in the root of the mathlib-port-status directory")
+    print("and have checked out mathlib under build/repos/mathlib.")
+    sys.exit(1)
+
+GITHUB_TOKEN_FILE = 'build/repos/github-token'
```
Comment (eric-wieser) [inline make_html.py]: I think it makes sense to load this from an env var, which is then easy to set in the workflow.yaml.

Also, if the var is missing I think you can just use `None`, and you'll just be rate-limited a little more

File: requirements.txt (hunk 3 above)
Comment (mo271) [reply requirements.txt]: remove this, also from requirements and also `logging` import

File: templates/index.j2 (hunk 4 above)
Comment (mo271) [reply templates/index.j2:52]: now sorting by the same column as before, which is now column 4

File: make_html.py (hunk 5 above)
Comment (mo271) [reply make_html.py]: reading from enviroment variable `GITHUB_TOKEN_FILE` now, otherwise use None....

File: make_html.py [hunk 6]
```diff
@@ -17,6 +19,17 @@
 
 from htmlify_comment import htmlify_comment
 
+
+github_token = open(os.environ["GITHUB_TOKEN_FILE"]).read().strip(
+) if "GITHUB_TOKEN_FILE" in os.environ else None
```
Comment (eric-wieser) [inline make_html.py]: Sorry, my suggestion is that this should actually be
```suggestion
github_token = os.environ.get("GITHUB_TOKEN")
```
as that's easier to configure with github actions

Comment (mo271) [reply make_html.py]: will adapt readme as well...

**Comment by eric-wieser** [comment]
> To get the labels just like they are on github one could copy the css from there, I suppose. Currently the yellow label is a bit hard to read, a dark font would be better here.

I'm happy to leave that to a follow-up. The github CSS seems to be doing some very complex things to decide what colors to use, and I think this is already a big win without that.

File: README.md [hunk 7]
```diff
@@ -1,2 +1,11 @@
 # mathlib-port-status
 Tools for managing the status of the port
+
+## Using the github api
+
+When getting an error "API rate limit exceeded for ...", you can [get a token](https://github.com/settings/tokens)
+and then
+```
+export GITHUB_TOKEN_FILE=path/to/my/token
```
Comment (eric-wieser) [inline README.md]:
```suggestion
export GITHUB_TOKEN=MY_TOKEN
```

File: README.md [hunk 8]
```diff
@@ -1,2 +1,11 @@
 # mathlib-port-status
 Tools for managing the status of the port
+
+## Using the github api
+
+When getting an error "API rate limit exceeded for ...", you can [get a token](https://github.com/settings/tokens)
+and then
+```
+export GITHUB_TOKEN=(cat path/to/my/github-token)
```
Comment (eric-wieser) [inline README.md]: Unless I'm missing something about bash,
```suggestion
export GITHUB_TOKEN=$(cat path/to/my/github-token)
```

File: README.md [hunk 9]
```diff
@@ -1,2 +1,11 @@
 # mathlib-port-status
 Tools for managing the status of the port
+
+## Using the github api
+
+When getting an error "API rate limit exceeded for ...", you can [get a token](https://github.com/settings/tokens)
+and then
+```
```
Comment (eric-wieser) [inline README.md]:
````suggestion
```bash
````

**Comment by mo271** [comment]
> > To get the labels just like they are on github one could copy the css from there, I suppose. Currently the yellow label is a bit hard to read, a dark font would be better here.
> 
> I'm happy to leave that to a follow-up. The github CSS seems to be doing some very complex things to decide what colors to use, and I think this is already a big win without that.

Sounds good to me, also this entire port status page will hopefully not be needed much longer...

File: make_html.py [hunk 10]
```diff
@@ -1,5 +1,6 @@
 from dataclasses import dataclass, field
 from enum import Enum
+import github
```
Comment (eric-wieser) [inline make_html.py]: This should be down by `jinja2`; the order is

* Builtin imports
* Third party imports
* Imports from this package

Comment (mo271) [reply make_html.py]: done

File: make_html.py [hunk 11]
```diff
@@ -8,6 +8,8 @@
 from typing import Optional, List
 import os
 
+
```
Comment (eric-wieser) [inline make_html.py]:
```suggestion
```

**Pull request merged**